### ISSUE TITLE
`dhall format` deletes comment

### ISSUE THREAD

**Issue by samuela** [dhall-lang/dhall-lang#1275]
If I have the following file: https://gist.github.com/samuela/b43b1eb3d7134512457fad3be76c94a3. And try running `dhall format foo.dhall`. The [comment](https://gist.github.com/samuela/b43b1eb3d7134512457fad3be76c94a3#file-example-dhall-L219-L222) here gets deleted.

Is this the intended behavior?

**Comment by basile-henry** [comment]
It is known behaviour. `dhall-haskell` currently only parses the comment at the top of the file and drops the rest.

There have been some efforts towards parsing all comments in order to not drop them in `dhall format` (like https://github.com/dhall-lang/dhall-haskell/pull/2173 which is now looking a bit outdated :sweat:)

**Comment by TristanCacqueray** [comment]
`dhall-haskell` also preserves the comment between `let` and identifiers and around record's attributes, e.g.:

```dhall
let -- | variable comment
  var = 42

 in {
  -- | attribute comment
  test = var
}
```

**Comment by samuela** [comment]
Thanks for explaining @basile-henry and @TristanCacqueray! I love the auto-format but it is very odd to me that it just deletes comments... perhaps it could just move them to the nearest let-comment or attribute comment?

**Comment by Gabriel439** [comment]
Yeah, we'd accept a pull request to do something like that.  It's just that I'm currently a bit busy at the moment to contribute this myself
